src/lib/comparison.ts

```diff
@@ -98,19 +131,58 @@ export class Comparison<Location> {
     const matchingFiles: Matches<Location> = new Map();
     const [ast, mapping] = await this.tokenizer.tokenizeFileWithMapping(file);
     for await (const { hash, location } of hashFilter.hashesFromString(ast)) {
-      const matches = this.index.get(hash);
-      if (matches) {
-        for (const [fileName, lineNumber] of matches) {
-          const match: [Location, Location] = [lineNumber, mapping[location]];
-          const lines = matchingFiles.get(fileName);
-          if (lines) {
-            lines.push(match);
-          } else {
-            matchingFiles.set(fileName, [match]);
+      if (!this.filteredHashSet.has(hash)) {
+        const matches = this.index.get(hash);
+        if (matches && this.filterByPassageCount(matches.length)) {
+          for (const [fileName, lineNumber] of matches) {
+            // add the match if the match is not with the file and if the file comes first when alphabetically sorted.
+            // This is done to avoid the duplicates in the following case: when file A matches with file B, file B will
```

Comment:
Our similarities are not symmetric, so I think we want to compare both A with B and B with A.
Example, file A contains 10 functions, file B contains only a single function which was copied from A. The similarities here are 10% in one direction and 100% in the other.